<a href="https://colab.research.google.com/github/Pavlenko-Kateryna/da_course/blob/main/hw_9_1_operations.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402
result = ((a + b) * c) / d
print(result)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [3]:
num_1 = 11283
num_2 = 89
result = num_1 % num_2
print(result)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [4]:
num_1 = 11283
num_2 = 89
result = num_1 // num_2
print(result)

126


4. Запишіть у змінну `name` ваше імʼя.

In [5]:
name = 'Katya'

5. Запишіть у змінну `surname` ваше прізвище.

In [6]:
surname = 'Pavlenko'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [7]:
name_surname = name + ' ' + surname
print(name_surname)

Katya Pavlenko


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [8]:
print(name_surname.upper())

KATYA PAVLENKO


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [9]:
print(name_surname[2])

t


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [10]:
print(surname[0])
print(surname[-1])

P
o


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [11]:
print(len(surname))

8


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [24]:
word = 'Antananarivo'
print(word.lower().count("a"))

4


In [25]:
word = 'Antananarivo'
print(word.upper().count("A"))

4


Очікувана відповідь: 4

Наступні вправи можуть здатись дещо складніше - вони для найуважніших!
Якщо не виходить - не страшно. Пишіть нам в чат, ми підкажемо!

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [12]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
result = my_string[3] == "The" and my_string[-4:] == "dog!"
print(result)

False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [21]:
split_str = my_string.split(" ")
result = "_".join(split_str)
print(result)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`